### PR TITLE
Suma payg rhui via payg client connection

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Support RHUI configuration via Pay-as-you-Go client connection in the cloud

Replace https://github.com/uyuni-project/uyuni/pull/7050
Port of https://github.com/SUSE/spacewalk/pull/21590

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- https://github.com/uyuni-project/uyuni-docs/pull/2274

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by parlt91 — APPROVED** [review]
Since this is a port of an already reviewed PR I am going to approve without full review. If you want a proper code review please re-request the review.

File: python/spacewalk/satellite_tools/reposync.py [hunk 1]
```diff
@@ -382,10 +382,11 @@ def verify_certificates_dates(certs):
         if is_cert:
             cert += line + '\n'
         if is_cert and line.startswith("-----END CERTIFICATE"):
+            is_cert = False
             if not verify_certificate_dates(cert):
-                return False
+                cert = ""
+                continue
```
Comment (agraul) [inline python/spacewalk/satellite_tools/reposync.py:388]: Do I understand this change correctly that we change the date verification from "all certs have valid dates" to "any cert has an valid date"? 

If that's the intention, we could also short-circuit the loop and return `True` as soon as we find the first valid cert.

**Review by agraul — COMMENTED** [review]
I just looked at the Python code and left a few comments. The biggest question mark for me is about the certificates and the changes around that. Is RHUI not setting a subject common name? Do we want to relax the dates check from "all" to "any"?

File: python/spacewalk/satellite_tools/satCerts.py [hunk 2]
```diff
@@ -39,7 +39,10 @@ def get_certificate_info(cert_str):
     not_before = cert.get_not_before().get_datetime()
     not_after = cert.get_not_after().get_datetime()
     subject = cert.get_subject()
-    cn = subject.CN
+    try:
+        cn = subject.CN
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply python/spacewalk/satellite_tools/satCerts.py:43]: A certificate may not have a common name. There are other parts for the subject used.

File: python/spacewalk/satellite_tools/satCerts.py [hunk 3]
```diff
@@ -39,7 +39,10 @@ def get_certificate_info(cert_str):
...
+    except:
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply python/spacewalk/satellite_tools/satCerts.py:44]: It is not easy to find out which one it is. It happens in one of the clouds, but we need to retest all of them.
I would suggest to keep this, and when you refactor reposync, we can maybe get rid of this whole paramater.
I think it is not needed, but I am not 100% sure.

File: python/spacewalk/satellite_tools/reposync.py (hunk 1 above)
Comment (mcalmer) [reply python/spacewalk/satellite_tools/reposync.py:388]: I have to run certificate bundles with this funktion.
And RedHat has some expired certificates in the bundle.
So if 1 cert is provided, it must be valid. If multiple are provided, it is ok if one has a valid date.

Comment (mcalmer) [reply python/spacewalk/satellite_tools/reposync.py:388]: I change this to return on the first valid certificate

File: python/spacewalk/satellite_tools/satCerts.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (agraul) [reply python/spacewalk/satellite_tools/satCerts.py:44]: I think it's always an `AttributeError`, just like when you try to access `subject.FOO`
```python
>>> cert = X509.load_cert_string(c)
>>> subject = cert.get_subject()
>>> subject.FOO
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.11/site-packages/M2Crypto/X509.py", line 359, in __getattr__
    raise AttributeError(self, attr)
AttributeError: (<M2Crypto.X509.X509_Name object at 0x7f22a0ba1bd0>, 'FOO')

File: python/spacewalk/satellite_tools/satCerts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (agraul) [reply python/spacewalk/satellite_tools/satCerts.py:43]: I tried to find it in the respective standard, but I don't know which standard actually defines this part :shrug: 

But even if this was specified as mandatory, we could allow missing CN fields...

File: python/spacewalk/satellite_tools/satCerts.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mcalmer) [reply python/spacewalk/satellite_tools/satCerts.py]: Ok, I have changed it

**Review by agraul — APPROVED** [review]
Python code LGTM!

**Pull request merged**